src/java.base/macosx/classes/sun/nio/fs/BsdFileSystem.java

```diff
@@ -71,13 +79,89 @@ public Set<String> supportedFileAttributeViews() {
         return SupportedFileFileAttributeViewsHolder.supportedFileAttributeViews;
     }
 
+    /**
+     * Clones the file whose path name is {@code src} to that whose path
+     * name is {@code dst} using the {@code clonefile} system call.
+     *
+     * @param src the path of the source file
+     * @param dst the path of the destination file (clone)
+     * @param followLinks whether to follow links
+     *
+     * @return 0 on success, IOStatus.UNSUPPORTED_CASE if the call does not work
+     *         with the given parameters, or IOStatus.UNSUPPORTED if cloning is
+     *         not supported on this platform
+     */
+    private int clone(UnixPath src, UnixPath dst, boolean followLinks)
+        throws IOException
+    {
+        // Do not clone unless both files are on the same volume and that
+        // volume is indicated as supporting file cloning
+        BsdFileStore bfs = (BsdFileStore)provider().getFileStore(src);
+        if (!bfs.equals(provider().getFileStore(dst.getParent())) ||
+            !bfs.supportsCloning())
```

Comment:
Is 'dst' a relative path, in which case you can't rely on getParent. In any case, I don't think we should be checking the FileStore here as that means finding the mount entry. If we do clonefile then I think you'll have to rely on the EXDEV error.

Reply:
Fixed in a7917d2216f111d2aa2d3190c927d5897c826da1.